pbsmetrics/go_metrics.go

```diff
@@ -313,8 +325,10 @@ func (me *Metrics) RecordRequest(labels Labels) {
 	}
 
 	// Handle the account metrics now.
-	am := me.getAccountMetrics(labels.PubID)
-	am.requestMeter.Mark(1)
+	if me.accountMetrics != nil {
+		am := me.getAccountMetrics(labels.PubID)
```

Comment:
We want plain account metrics, just not account*adapter metrics.

Reply:
Got confused, I thought we wanted to disable all of the metrics this function works with. Which ones do you want us to keep?
```
255 // getAccountMetrics gets or registers the account metrics for account "id".
256 // There is no getBlankAccountMetrics() as all metrics are generated dynamically.
257 func (me *Metrics) getAccountMetrics(id string) *accountMetrics {
258     var am *accountMetrics
259     var ok bool
260
261     me.accountMetricsRWMutex.RLock()
262     am, ok = me.accountMetrics[id]
263     me.accountMetricsRWMutex.RUnlock()
264
265     if ok {
266         return am
267     }
268
269     me.accountMetricsRWMutex.Lock()
270     // Made sure to use defer as we have two exit points: we want to unlock the mutex as quickly as possible.
271     defer me.accountMetricsRWMutex.Unlock()
272
273     am, ok = me.accountMetrics[id]
274     if ok {
275         // Unlock and return as quickly as possible
276         return am
277     }
278     am = &accountMetrics{}
279     am.requestMeter = metrics.GetOrRegisterMeter(fmt.Sprintf("account.%s.requests", id), me.MetricsRegistry)
280     am.bidsReceivedMeter = metrics.GetOrRegisterMeter(fmt.Sprintf("account.%s.bids_received", id), me.MetricsRegistry)
281     am.priceHistogram = metrics.GetOrRegisterHistogram(fmt.Sprintf("account.%s.prices", id), me.MetricsRegistry, metrics.NewExpDecaySample(1028, 0.015))
282     am.adapterMetrics = make(map[openrtb_ext.BidderName]*AdapterMetrics, len(me.exchanges))
283     for _, a := range me.exchanges {
284         am.adapterMetrics[a] = makeBlankAdapterMetrics()
285         registerAdapterMetrics(me.MetricsRegistry, fmt.Sprintf("account.%s", id), string(a), am.adapterMetrics[a])
286     }
287
288     me.accountMetrics[id] = am
289
290     return am
291 }
pbsmetrics/go_metrics.go [+]
```

Reply:
the ones in the for loop in line 283? Like this:
```
     +  if !AdapterMetricsDisabled {
283         for _, a := range me.exchanges {
284             am.adapterMetrics[a] = makeBlankAdapterMetrics()
285             registerAdapterMetrics(me.MetricsRegistry, fmt.Sprintf("account.%s", id), string(a), am.adapterMetrics[a])
286         }
     +  }
pbsmetrics/go_metrics.go [+]
```

Reply:
Yes, we don't want to do anything with `adapterMetrics` inside the account metrics context. There are adapter metrics elsewhere that do not reference account.

Reply:
Got it. Update corrects this issue